sarracen/readers/read_shamrock.py

```diff
@@ -0,0 +1,71 @@
+from typing import Union
+
+import numpy as np
+import pandas as pd
+
+import vtk
+from vtk.util.numpy_support import vtk_to_numpy
+
+from ..sarracen_dataframe import SarracenDataFrame
+
+def read_vtk(filename, pmass):
+    """
+    Read data from a SHAMROCK vtk file ('big' simulation current format).
+
+    Parameters
+    ----------
+    filename : str
+        Name of the file to be loaded.
+    pmass : float
+        Mass of particles in the simulation (for now, it is assumed all particles)
+        have the same mass).
+
+    Returns
+    -------
+    SarracenDataFrame
+
+    """
+
+    # read the vtk file
+    reader = vtk.vtkDataSetReader()
+    reader.SetFileName(filename)
+    reader.Update()
+    vtk_data = reader.GetOutput()
+    num_arrays = vtk_data.GetPointData().GetNumberOfArrays()
+
+    # initialize the dataframe
+    df = pd.DataFrame()
+
+    for i in range(num_arrays):
+        vtk_array = vtk_data.GetPointData().GetArray(i)
+        array_name = vtk_array.GetName()
+        numpy_array = vtk_to_numpy(vtk_array)
+        ndim = numpy_array.ndim
+
+        if ndim==1:
+            print(array_name)
+            df[array_name] = numpy_array
+
+        else:
+            df[array_name + 'x'] = numpy_array[:, 0]
+            df[array_name + 'y'] = numpy_array[:, 1]
+            df[array_name + 'z'] = numpy_array[:, 2]
+
+    # add B
+    df['Bx'] = df['B/rhox'] * df['rho']
+    df['By'] = df['B/rhoy'] * df['rho']
+    df['Bz'] = df['B/rhoz'] * df['rho']
```

Comment:
Sorry just another comment I should have realized before now -- what if the file doesn't contain magnetic field? I can imagine that you would often pure hydro calculations.

A simple check would be:
`if 'B/rhox' in df.columns:`
`    df['Bx'] = df['B/rhox'] * df['rho']`